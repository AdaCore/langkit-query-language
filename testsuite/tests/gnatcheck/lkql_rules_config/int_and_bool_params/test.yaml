driver: gnatcheck
control:
  - [SKIP, "mode == 'ada'", "LKQL rule file feature unavailable on Ada implementation"]
format: brief
input_sources:
 - main.adb

tests:
 - label: Numeric_Literals
   lkql_rule_file: num_lit.lkql
 - label: Others_In_Case_Statements
   lkql_rule_file: others_in_case.lkql
 - label: Membership_Tests
   lkql_rule_file: membership_tests.lkql
 - label: Invalid type
   lkql_rule_file: invalid_type.lkql
