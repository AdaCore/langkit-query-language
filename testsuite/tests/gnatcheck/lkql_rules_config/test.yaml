driver: gnatcheck
project: prj.gpr
lkql_rule_file: config.lkql
format: brief
show_rule: True
control:
  - [SKIP, "mode == 'ada'", "LKQL rule file feature unavailable on Ada implementation"]