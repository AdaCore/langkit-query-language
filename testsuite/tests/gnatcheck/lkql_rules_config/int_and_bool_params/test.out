Numeric_Literals
================

main.adb:4:11: rule violation: numeric literal

Others_In_Case_Statements
=========================

main.adb:11:12: rule violation: OTHERS choice in case statement

Membership_Tests
================

main.adb:17:10: rule violation: membership test

Invalid type
============

gnatcheck: error: invalid parameter for rule "membership_tests": 'float_types_only' value should be a boolean (invalid_type.lkql)
gnatcheck: error: no rule to check specified
try "gnatcheck --help" for more information.
>>>program returned status code 2
